core/src/main/java/org/infinispan/util/concurrent/CompletionStages.java

```diff
@@ -198,6 +202,28 @@ R getValue() {
       }
    }
 
+   private static class OrBooleanAggregateCompletionStage extends AbstractAggregateCompletionStage<Boolean> {
+
+      private volatile boolean value = false;
+
+      @Override
+      Boolean getValue() {
+         return value;
+      }
+
+      @Override
+      public void accept(Object o, Throwable t) {
+         if (t != null) {
+            super.accept(null, t);
+            return;
+         }
+         if (o instanceof Boolean && (Boolean) o) {
```

Comment:
I was debating about making the AggregateCompletionStage use generics in a better way. I am kinda thinking we may want to finally do that if we start adding more impls.

Also I do slightly worry about having more implementations since right now we only have 2.

In the past I used the value aggregate completion stage with an atomic variable to do this type of work. See https://github.com/infinispan/infinispan/blob/main/core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java#L679

But this way may be better, hard to tell.

Reply:
well, at the end of the day, the difference is minimal; it just avoids allocating an `AtomicBoolean`. 
In my personal opinion, I don't like to have to create an `AtomicBoolean` and create other lambdas to update it. 

The only small thing that anoyed me was having
```java
AggregateCompletionStage<R> dependsOn(CompletionStage<?> stage);
``` 
instead of 
```java
AggregateCompletionStage<R> dependsOn(CompletionStage<R> stage);
// or
AggregateCompletionStage<R> dependsOn(CompletionStage<? extends R> stage);
```

Reply:
Yeah, that is what I meant by fixing the generics. The more I think about this case wont have issues with the different impls as we should always be using the same on in various locations. So I am good with how it is. I will create a JIRA to change the generics over.